feature-cancel-insurance/src/main/kotlin/com/hedvig/android/cancelinsurance/ui/CancelInsuranceScreen.kt

```diff
@@ -0,0 +1,159 @@
+import com.hedvig.android.core.designsystem.component.datepicker.HedvigDatePicker
+import com.hedvig.android.core.designsystem.material3.HedvigMaterial3Theme
+import com.hedvig.android.core.ui.appbar.m3.TopAppBarWithBack
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CancelInsuranceScreen(
+  datePickerState: DatePickerState,
+  dateValidator: (Long) -> Boolean,
+  canSubmit: Boolean,
+  submit: () -> Unit,
+  hasError: Boolean,
+  showedError: () -> Unit,
+  navigateBack: () -> Unit,
+) {
+  Box(Modifier.fillMaxSize()) {
+    Column {
+      val topAppBarScrollBehavior = TopAppBarDefaults.pinnedScrollBehavior()
+      TopAppBarWithBack(
+        onClick = navigateBack,
+        title = "Set termination date",
+        scrollBehavior = topAppBarScrollBehavior,
+      )
+      Column(
+        Modifier
+          .fillMaxSize()
+          .nestedScroll(topAppBarScrollBehavior.nestedScrollConnection)
+          .verticalScroll(rememberScrollState())
+          .windowInsetsPadding(WindowInsets.safeDrawing.only(WindowInsetsSides.Horizontal)),
+      ) {
+        Spacer(Modifier.height(20.dp))
+        ChatCard()
+        Spacer(Modifier.height(20.dp))
+        Spacer(Modifier.weight(1f))
+        DatePickerCard(datePickerState, dateValidator)
+        Spacer(Modifier.height(16.dp))
+        LargeContainedTextButton(
+          text = stringResource(hedvig.resources.R.string.general_continue_button),
+          onClick = submit,
+          enabled = canSubmit,
+          modifier = Modifier.padding(horizontal = 16.dp),
+        )
+        Spacer(Modifier.height(16.dp))
+        Spacer(
+          Modifier
+            .windowInsetsPadding(
+              WindowInsets.safeDrawing.only(WindowInsetsSides.Bottom),
+            ),
+        )
+      }
+    }
+    ErrorSnackbar(
+      hasError = hasError,
+      showedError = showedError,
+      modifier = Modifier
+        .align(Alignment.BottomCenter)
+        .windowInsetsPadding(WindowInsets.safeDrawing),
+    )
+  }
+}
+
+@Composable
+private fun ChatCard() {
+  HedvigCard(
+    shape = RoundedCornerShape(12.dp),
+    elevation = HedvigCardElevation.Elevated(),
+    modifier = Modifier.padding(start = 16.dp, end = 32.dp),
+  ) {
+    Text(
+      text = "Please set termination date for your insurance.",
+      modifier = Modifier.padding(vertical = 12.dp, horizontal = 16.dp),
+    ) // TODO Add parameter for insurance name
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DatePickerCard(
+  datePickerState: DatePickerState,
+  dateValidator: (Long) -> Boolean,
+) {
+  HedvigMaterial3Theme {
+    HedvigCard(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(horizontal = 16.dp),
+    ) {
+      HedvigDatePicker(
+        datePickerState = datePickerState,
+        dateValidator = dateValidator,
+      )
+    }
+  }
+}
+
+@Composable
+fun ErrorSnackbar(hasError: Boolean, showedError: () -> Unit, modifier: Modifier = Modifier) {
```

Comment:
Can we extract this so that we can reuse it? Or do we have something similar we can use already, like a dialog?

Reply:
Extracted it here https://github.com/HedvigInsurance/android/pull/1434
For this one I used a snackbar since we don't really have a meaningful message to show to the member about what's wrong. In general we don't have any guidelines on how to handle such cases, so I think either approach would be fine.
The Dialog is one that requires them to actively dismiss it and is kinda in your face, while just saying "Something went wrong", so I thought in this case the snackbar would suffice. Don't have strong feelings about this though, if anything I'd love to discuss with a designer how such bad scenarios should be handled in this screen but alas we can't do that atm